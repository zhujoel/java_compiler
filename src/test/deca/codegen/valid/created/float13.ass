; Test pour savoir si la pile est pleine
	TSTO #13
	BOV pile_pleine
	ADDSP #11
; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 3(GB)
	LOAD code.Object.equals, R0
	STORE R0, 5(GB)
; Main Function
; Declaration des variables
	LOAD #0x1.266666p1, R2
	STORE R2, 7(GB)
	LOAD #1, R2
	STORE R2, 9(GB)
; Beginning of main instructions:
; +
	LOAD 7(GB), R2
	LOAD 9(GB), R3
	FLOAT R3, R3
	ADD R3, R2
	BOV debordement
	STORE R2, 7(GB)
	LOAD 7(GB), R1
	WFLOAT
	HALT
code.Object.equals:
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
