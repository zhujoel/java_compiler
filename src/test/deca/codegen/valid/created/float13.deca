{
    float a = 2.3;
    int b = 1;
    a = a + b;
    print(a);
}
