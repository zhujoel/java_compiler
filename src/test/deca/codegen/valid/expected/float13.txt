3.30000e+00
